source/weather.py

```diff
@@ -0,0 +1,265 @@
+# Copyright 2021, Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Representations and conversions of the data returned by the weather API."""
+from datetime import time, timedelta
+
+from mycroft.util.time import now_local
+from .config import MILES_PER_HOUR
+from .util import convert_to_local_datetime
+
+THIRTY_PERCENT = 30
+SATURDAY = 5
+SUNDAY = 6
+WIND_DIRECTION_CONVERSION = (
+    (22.5, "north"),
+    (67.5, "northeast"),
+    (112.5, "east"),
+    (157.5, "southeast"),
+    (202.5, "south"),
+    (247.5, "southwest"),
+    (292.5, "west"),
+    (337.5, "northwest"),
+)
+
+
+def is_daily_forecast(weather_report) -> bool:
+    """Helper function to determine if the object passed is a DailyWeather object."""
+    return isinstance(weather_report, DailyWeather)
+
+
+def is_hourly_forecast(weather_report) -> bool:
+    """Helper function to determine if the object passed is a HourlyWeather object."""
+    return isinstance(weather_report, HourlyWeather)
+
+
+def is_current_weather(weather_report) -> bool:
+    """Helper function to determine if the object passed is a CurrentWeather object."""
+    return isinstance(weather_report, CurrentWeather)
+
+
+class WeatherCondition:
+    """Data representation of a weather conditions JSON object from the API"""
+
+    def __init__(self, conditions: dict):
+        self.id = conditions["id"]
+        self.category = conditions["main"]
+        self.description = conditions["description"]
+        self.icon = conditions["icon"]
+
+
+class Weather:
+    """Abstract data representation of commonalities in forecast types."""
+
+    def __init__(self, weather: dict, timezone: str):
+        self.date_time = convert_to_local_datetime(weather["dt"], timezone)
+        self.feels_like = weather["feelsLike"]
+        self.pressure = weather["pressure"]
+        self.humidity = weather["humidity"]
+        self.dew_point = weather["dewPoint"]
+        self.clouds = weather["clouds"]
+        self.wind_speed = int(weather["windSpeed"])
+        self.wind_direction = self._determine_wind_direction(weather["windDeg"])
+        self.condition = WeatherCondition(weather["weather"][0])
+
+    @staticmethod
+    def _determine_wind_direction(degree_direction: int):
```

Comment:
should the type spec be (int, float) (or however a number is specified)

Reply:
I am not sure I understand.  The API docs show degrees expressed in an integer format.

Reply:
I mean a float would work just as well and should be an acceptable type?

Reply:
Yes, float would work just as well.  But in its current iteration, it will never get a float.  I feel like adding that flexibility would be confusing.  It can always be added later if someone adds a new API that returns a float.